src/lexer/lexer.py

```diff
@@ -0,0 +1,117 @@
+"""
+Lexer module used by PLY's lexer-generator.
+"""
+# pylint: disable=missing-function-docstring, invalid-name
+
+from ply.lex import lex, TOKEN
+from lexer.token_types import TokenType
+from utils.console_io import default_console_io
+
+
+class Lexer:
+    """Lexer using PLY for tokenizing an input stream."""
+
+    # Maps <keyword in input> to <token type>
+    reserved_words = {
+        "to": TokenType.TO,
+        "miten": TokenType.TO,
+        "end": TokenType.END,
+        "valmis": TokenType.END,
+        "fd": TokenType.FD,
+        "forward": TokenType.FD,
+        "et": TokenType.FD,
+        "eteen": TokenType.FD,
+        "bk": TokenType.BK,
+        "backward": TokenType.BK,
+        "ta": TokenType.BK,
+        "taakse": TokenType.BK,
+        "rt": TokenType.RT,
+        "right": TokenType.RT,
+        "oi": TokenType.RT,
+        "oikealle": TokenType.RT,
+        "lt": TokenType.LT,
+        "left": TokenType.LT,
+        "va": TokenType.LT,
+        "vasemmalle": TokenType.LT,
+        "stop": TokenType.STOP,
+        "seis": TokenType.STOP,
+        "make": TokenType.MAKE,
+        "olkoon": TokenType.MAKE,
+        "if": TokenType.IF,
+        "jos": TokenType.IF,
+        "for": TokenType.FOR,
+        "luvuille": TokenType.FOR,
```

Comment:
Onko "luvuille" ainut vaihtoehto for-silmukan nimelle tässä tapauksessa? Tarkoitan siis sitä, että onko Logossa aina kyse numeroiden silmukoinnista? Jos on, niin tämä on ihan ok.